src/Controllers/Common/ContentController.cs

```diff
@@ -778,9 +778,16 @@ public IActionResult GetAvatarByUserId([FromForm] Guid userId)
 
     [HttpPost]
     [Produces("application/xml")]
-    [Route("ContentWebService.asmx/GetUnselectedPetByTypes")] // used by old SoD (e.g. 1.13)
+    [Route("ContentWebService.asmx/GetUnselectedPetByTypes")] // used by old SoD (e.g. 1.13) and Math Blaster
     [VikingSession(UseLock=false)]
-    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string petTypeIDs, [FromForm] bool active) {
+    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string? apiKey, [FromForm] string? userId, [FromForm] string petTypeIDs, [FromForm] bool active) {
+        if (apiKey != null
+            && (ClientVersion.GetVersion(apiKey) & ClientVersion.MB) != 0
```

Comment:
`ClientVersion.GetVersion(apiKey) == ClientVersion.MB` ... this is not real bitflag/bitmask value and using `&` need some extra checks  (like `ClientVersion.GetVersion(apiKey) & ClientVersion.MB == ClientVersion.MB` or `ClientVersion.GetVersion(apiKey) <= ClientVersion.Max_OldJS`)